marconi-chain-index/src/Marconi/ChainIndex/Indexers/ExtLedgerStateCoordinator.hs

```diff
@@ -0,0 +1,757 @@
+module Marconi.ChainIndex.Indexers.ExtLedgerStateCoordinator (
+  -- * Types
+  ExtLedgerState,
+  ExtLedgerStateEvent (..),
+  ExtLedgerConfig,
+  ExtLedgerStateCoordinator (ExtLedgerStateCoordinator),
+  ExtLedgerStateCoordinatorConfig (ExtLedgerStateCoordinatorConfig),
+  ExtLedgerStateWorkerConfig (..),
+
+  -- * Constructors
+  mkExtLedgerStateCoordinator,
+  extLedgerStateWorker,
+
+  -- * Utils
+  newEpochPreprocessor,
+  readGenesisFile,
+) where
+
+import Cardano.Api qualified as C
+import Cardano.Api.Extended qualified as CE
+import Cardano.Api.Shelley qualified as C
+import Cardano.BM.Trace qualified as BM
+import Cardano.BM.Tracing qualified as BM
+import Cardano.Ledger.Shelley.API qualified as Ledger
+import Codec.CBOR.Decoding qualified as CBOR
+import Codec.CBOR.Encoding qualified as CBOR
+import Codec.CBOR.Read qualified as CBOR
+import Codec.CBOR.Write qualified as CBOR
+import Control.Arrow ((<<<))
+import Control.Concurrent qualified as Con
+import Control.Exception (throw)
+import Control.Lens (Lens', (%~), (&), (-~), (.=), (.~), (^.))
+import Control.Lens qualified as Lens
+import Control.Monad (foldM, (<=<))
+import Control.Monad.Except (ExceptT, MonadError (throwError), MonadTrans (lift), runExceptT)
+import Control.Monad.IO.Class (MonadIO (liftIO))
+import Control.Monad.State.Strict (MonadState (put), gets)
+import Control.Monad.Trans.Maybe (MaybeT (runMaybeT))
+import Data.Bifunctor (Bifunctor (bimap))
+import Data.ByteString qualified as BS
+import Data.ByteString.Base16 qualified as Base16
+import Data.ByteString.Lazy qualified as BS.Lazy
+import Data.ByteString.Short qualified as BS.Short
+import Data.Data (Proxy (Proxy))
+import Data.List (sortOn)
+import Data.Map qualified as Map
+import Data.Maybe (fromMaybe, listToMaybe)
+import Data.Ord (Down (Down), comparing)
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Data.Text.Encoding qualified as Text
+import Marconi.Cardano.Core.Extract.WithDistance (WithDistance, chainDistance)
+import Marconi.Cardano.Core.Orphans ()
+import Marconi.Cardano.Core.Types (BlockEvent, SecurityParam (SecurityParam), blockInMode)
+import Marconi.Core (IsSync (lastStablePoint))
+import Marconi.Core qualified as Core
+import Marconi.Core.Preprocessor qualified as Core
+import Ouroboros.Consensus.Cardano.Block qualified as O
+import Ouroboros.Consensus.Config qualified as O
+import Ouroboros.Consensus.Ledger.Extended qualified as O
+import Ouroboros.Consensus.Node.NetworkProtocolVersion qualified as O
+import Ouroboros.Consensus.Node.Serialisation qualified as O
+import Ouroboros.Consensus.Shelley.Ledger qualified as O
+import Ouroboros.Consensus.Storage.Serialisation qualified as O
+import Prettyprinter (Pretty (pretty))
+import System.Directory (createDirectoryIfMissing)
+import System.FilePath ((</>))
+import Text.Read qualified as Text
+
+type ExtLedgerState = O.ExtLedgerState (O.HardForkBlock (O.CardanoEras O.StandardCrypto))
```

Comment:
Oh this is where the EUTxOs are? So the modelling of this data is in the Ouroboros repository, right?

Reply:
Yes. :)